### ISSUE TITLE
NullPointerException encountered with code that has try block in constructor.

### ISSUE THREAD

**Issue by coollog** [uber/NullAway#133]
Using the Gradle plugin:

```groovy
plugins {
    id 'net.ltgt.apt' version '0.13'
    id 'net.ltgt.errorprone' version '0.0.13'
}

dependencies {
    apt 'com.uber.nullaway:nullaway:0.3.5'
    errorprone 'com.google.errorprone:error_prone_core:2.2.0'
}
```

Example code that breaks (any try block without finally in a constructor):

```groovy
class SomeClass {
  SomeClass() {
    try (Stream<Integer> stream = Stream.of(1, 2, 3)) {}
  }
}
```

Head of stacktrace:

```
java.lang.NullPointerException
        at com.uber.nullaway.NullAway.getSafeInitMethods(NullAway.java:1267)
        at com.uber.nullaway.NullAway.getSafeInitMethods(NullAway.java:1282)
        at com.uber.nullaway.NullAway.guaranteedNonNullForConstructor(NullAway.java:1196)
        at com.uber.nullaway.NullAway.checkConstructorInitialization(NullAway.java:1176)
        at com.uber.nullaway.NullAway.checkFieldInitialization(NullAway.java:1053)
        at com.uber.nullaway.NullAway.matchClass(NullAway.java:896)
```

I believe this was introduced in https://github.com/uber/NullAway/pull/123.

**Comment by msridhar** [comment]
Thanks for the report!  @lazaroclapp do you have time to take a quick look?

**Comment by lazaroclapp** [comment]
Hi @coollog ! Thanks for the bug report! Indeed we weren't considering the cases for try-with-resources blocks. Working on #135 to fix this issue!

**Comment by coollog** [comment]
Thanks for the quick fix! Would this be in v0.3.6 soon?

**Comment by coollog** [comment]
Great, thanks!
